control/src/main/java/org/kaazing/k3po/control/internal/Control.java

```diff
@@ -247,8 +281,11 @@ private PreparedEvent readPreparedEvent() throws IOException {
                 case "name":
                     // compatibility
                     break;
+                case "barrier":
+                    prepared.getBarriers().add(headerValue);
+                    break;
                 default:
-                    throw new IllegalStateException("Unrecognized event header: " + headerName);
+                    // NOP allow unrecognized headers for future compatibility
```

Comment:
I'll mess around with doing default java logging and see how junit deals with that.